at_commons/lib/src/exception/exception_stack.dart

```diff
@@ -0,0 +1,40 @@
+import 'dart:collection';
+
+import 'package:at_commons/src/exception/at_exceptions.dart';
+
+class AtExceptionStack implements Comparable {
+  final _exceptionList = Queue<Triple>();
+
+  void add(Triple triple) {
+    if (compareTo(triple) == 0) {
+      return;
+    }
+    _exceptionList.addFirst(triple);
+  }
+
+  String getTraceMessage() {
+    String fullMessage = '';
+    for (Triple element in _exceptionList) {
+      fullMessage += '${element.message}\n';
+    }
+    return fullMessage;
+  }
+
+  @override
+  int compareTo(triple) {
+    for (var element in _exceptionList) {
+      if (element.message == triple.message) {
+        return 0;
+      }
+    }
+    return 1;
+  }
+}
+
+class Triple {
```

Comment:
I *really* don't like this class name :-) Can you come up with a name that gives a little more of a clue as to what it is?

Reply:
Renamed class Triple as AtChainedException